### PR TITLE
Update Environment variable precedence docs

### PR DESCRIPTION
**Pull request by jeremybumsted**
Adds a section in environment variable precedence to add clarification around multiple pipeline uploads.

### CODE REVIEW

File: pages/pipelines/environment_variables.md [hunk 1]
```diff
@@ -255,3 +255,11 @@ environment, and will override any previous values that are set.
 This is the environment your command runs in 🎉
 
 Finally, if your job's commands make any changes to the environment, those changes will only survive as long as the script is running.
+
+### Environment variables in multiple pipeline uploads
+
+In the event that you are uploading multiple pipeline definitions, depending on the order of uploads, environment variable precedence behaves differently.
```
Comment (mbelton-buildkite) [inline pages/pipelines/environment_variables.md:261]: General tightening:

```suggestion
If you upload multiple pipeline definitions, the environment variable precedence behaves differently depending on the order of uploads.
```

File: pages/pipelines/environment_variables.md [hunk 2]
```diff
@@ -255,3 +255,11 @@ environment, and will override any previous values that are set.
...
+
+If you are uploading pipeline definitions concurrently the value of the environment variable will match the uploading pipeline definition.
+,
+Pipeline uploads performed serially will result in the most recent declaration of an environment variable to be true.
```
Comment (mbelton-buildkite) [inline pages/pipelines/environment_variables.md:265]: I think:

```suggestion
When you upload pipeline definitions in serial, environment variables have the value from the most recent declaration.
```

Is that right?

File: pages/pipelines/environment_variables.md [hunk 3]
```diff
@@ -255,3 +255,11 @@ environment, and will override any previous values that are set.
 This is the environment your command runs in 🎉
 
 Finally, if your job's commands make any changes to the environment, those changes will only survive as long as the script is running.
+
+### Environment variables in multiple pipeline uploads
+
+In the event that you are uploading multiple pipeline definitions, depending on the order of uploads, environment variable precedence behaves differently.
+
+If you are uploading pipeline definitions concurrently the value of the environment variable will match the uploading pipeline definition.
+,
```
Comment (mbelton-buildkite) [inline pages/pipelines/environment_variables.md:264]: Typo?

```suggestion

```

File: pages/pipelines/environment_variables.md [hunk 4]
```diff
@@ -255,3 +255,11 @@ environment, and will override any previous values that are set.
 This is the environment your command runs in 🎉
 
 Finally, if your job's commands make any changes to the environment, those changes will only survive as long as the script is running.
+
+### Environment variables in multiple pipeline uploads
+
+In the event that you are uploading multiple pipeline definitions, depending on the order of uploads, environment variable precedence behaves differently.
+
+If you are uploading pipeline definitions concurrently the value of the environment variable will match the uploading pipeline definition.
```
Comment (mbelton-buildkite) [inline pages/pipelines/environment_variables.md:263]: Can you please unpack this one a bit more for me? Is it deterministic?

Comment (jeremybumsted) [reply pages/pipelines/environment_variables.md:263]: @mbelton-buildkite apologies for the delay there. Yes, it's deterministic.

If we use this example, where we upload the pipeline definitions concurrently:
```yaml
steps:
  - command: |
      buildkite-agent pipeline upload .buildkite/p1.yml
      buildkite-agent pipeline upload .buildkite/p2.yml
```
**p1.yml**

```yaml
env:
  FOO: "bar"

steps:
  - label: "P1" 
    command: echo "+++ p1 - $$FOO"
``` 

**p2.yml**
```yaml
env:
  FOO: "baz"

steps:
  - command: echo "+++ p2 - $$FOO"
```

The resulting pipeline will look like this:

image

The value of `FOO` will match that of the declaration of `FOO` in the respective uploading pipeline YAML.

**Comment by jeremybumsted** [comment]
After some more noodling, I don't think it actually makes enough sense to try and put this into the docs - I think it is actually explained well enough here: https://buildkite.com/docs/pipelines/environment-variables#environment-variable-precedence

**Pull request closed without merging**